### PR TITLE
[Feature](Prepared Statement) fix and enable enable_server_side_prepared_statement by default #36581

### PR DESCRIPTION
**Pull request by eldenmoon**
picked from #36581

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: regression-test/suites/insert_p0/insert_group_commit_with_exception.groovy [hunk 1]
```diff
@@ -161,9 +161,15 @@ suite("insert_group_commit_with_exception") {
                 if (item == "nereids") {
                     statement.execute("set enable_nereids_dml = true;");
                     statement.execute("set enable_nereids_planner=true;");
+<<<<<<< HEAD
```
Comment (xiaokang) [inline regression-test/suites/insert_p0/insert_group_commit_with_exception.groovy:164]: fix it

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: regression-test/suites/insert_p0/insert_group_commit_with_exception.groovy (hunk 1 above)
Comment (eldenmoon) [reply regression-test/suites/insert_p0/insert_group_commit_with_exception.groovy]: done

**Pull request merged**